### PR TITLE
Run Espresso tests on Travis

### PR DESCRIPTION
**Pull request by rocboronat**
Another way to implement #80 

In this case, we just add some noise to Travis log with the flags `--stacktrace --info`. This should make that Travis don't kill our process due to not writing into the log.

### CODE REVIEW

**Comment by rocboronat** [comment]
Hey @ferranpons ! It seems that this approach works as expected. By the way, the emulator is really, really slow passing tests... Let's see if the build ends :·)

**Comment by rocboronat** [comment]
Methods added at #78 fail at the emulator :·( Let's try a simple `@AllowFlaky`...

**Comment by rocboronat** [comment]
Well, as is, it works on my localhost, on emulators 19 and 21. Let's see what happens at Travis. If it crashes, I'll try with `@AllowFlaky` at `clickMenu()` tests.

**Comment by rocboronat** [comment]
It works! :tada:

**Comment by Sloy** [comment]
Why do we need to ignore menu tests?

**Comment by rocboronat** [comment]
Let me forward you to the Travis log... but with some links to the exact lines ;·) Let fix that test on another PR.

checkMenuOnOverflowClick_byText() https://travis-ci.org/SchibstedSpain/Barista/builds/237567295#L7714

checkMenuOnOverflowClick_byId() https://travis-ci.org/SchibstedSpain/Barista/builds/237567295#L7718

checkMenuOnOverflowClick_byText() https://travis-ci.org/SchibstedSpain/Barista/builds/237567295#L7729

checkMenuOnOverflowClick_byId() https://travis-ci.org/SchibstedSpain/Barista/builds/237567295#L7733

**Comment by rocboronat** [comment]
Some feedback about this, @Sloy ? And what do you think about the annotation's name?

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/FlakyRuleTest.java [hunk 1]
```diff
@@ -51,7 +51,9 @@ public void someFlakyTest() throws Exception {
     }
   }
 
+  // WARNING: this test is flaky. Avoid running it at the C.I.
   @Test
+  @Ignore
```
Comment (Sloy) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/FlakyRuleTest.java]: Don't you want to use your new annotation here? From the comment it looks like you do.

**Comment by Sloy** [comment]
Well... the log is sooo long and big that the link to the exact line doesn't work 😹 
But other than that, I don't know :/ It's weird that only the menu tests are failing. I don't feel very confident.

The annotation name looks very explanatory, so that's OK.

**Comment by Sloy** [comment]
Humm actually the tests that fail are the ones that open the overflow menu. Maybe it's something with the Android version?
Did you try running the tests locally in the same emulator version than Travis? Or with the same parameters? Maybe the `-no-window -gpu off -no-boot-anim` have something to do with this...

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/FlakyRuleTest.java (hunk 1 above)
Comment (rocboronat) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/FlakyRuleTest.java]: Nope. This test also crashes on localhost, so I prefer to keep it ignored on Travis AND in our computers. Newcomers doesn't know what's expected from this test, so it's strange to understand that a red is something randomly expected from this test. In addition, that's not related to passing the tests on Travis.

**Comment by rocboronat** [comment]
@Sloy what do you think about fixing those tests in another PR? This PR runs 124 tests of 130 on Travis. It's the step forward that we expected since January 😅

**Comment by Sloy** [comment]
@rocboronat sure, let's go ahead 👍

**Pull request merged**